From d6fd2757f7aab638022ffa635e32e21594ec382d Mon Sep 17 00:00:00 2001
From: Stefan Weil <sw@weilnetz.de>
Date: Sat, 21 Jan 2023 20:36:37 +0100
Subject: [PATCH 05/28] iw: event: fix printf format error

tv_usec can be a 64 bit integer which causes a compiler warning:

event.c: In function 'print_event':
event.c:930:41: warning: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type 'suseconds_t' {aka 'long long int'} [-Wformat=]
  930 |                         printf("[%s.%06lu]: ", buf, args->ts.tv_usec);
      |                                     ~~~~^           ~~~~~~~~~~~~~~~~
      |                                         |                   |
      |                                         long unsigned int   suseconds_t {aka long long int}
      |                                     %06llu

Signed-off-by: Stefan Weil <sw@weilnetz.de>
Link: https://lore.kernel.org/r/20230121193637.347109-1-sw@weilnetz.de
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 event.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/event.c b/event.c
index 4c37297..fa2e125 100644
--- a/event.c
+++ b/event.c
@@ -942,7 +942,7 @@ static int print_event(struct nl_msg *msg, void *arg)
 
 			memset(buf, 0, 255);
 			strftime(buf, sizeof(buf), "%Y-%m-%d %H:%M:%S", tm);
-			printf("[%s.%06lu]: ", buf, args->ts.tv_usec);
+			printf("[%s.%06lu]: ", buf, (unsigned long )args->ts.tv_usec);
 		} else {
 			printf("%llu.%06llu: ", usecs/1000000, usecs % 1000000);
 		}
-- 
2.39.2

